commit da0bf27930ce4add54a58dae56d6c76948898cac
Author: Richard H Lee <ricardohenrylee@gmail.com>
Date:   Mon May 9 22:28:58 2016 +0100

ass_process_chunk: Use llrint, not lrint

libass's ass_process_chunk expects long long int for the timecode and
durations arguments, thus should use llrint instead of lrint.

This does not cause any problems on most platforms, but on cygwin, it
causes strange subtitle behaviour, like subtitles not showing, getting
stuck or old subtitles showing at the same time as new subtitles.
---

diff --git a/sub/sd_ass.c b/sub/sd_ass.c
index ebe04a2..6f35053 100644
--- a/sub/sd_ass.c
+++ b/sub/sd_ass.c
@@ -270,8 +270,8 @@ static void decode(struct sd *sd, struct demux_packet *packet)
         // Note that for this packet format, libass has an internal mechanism
         // for discarding duplicate (already seen) packets.
         ass_process_chunk(track, packet->buffer, packet->len,
-                          lrint(packet->pts * 1000),
-                          lrint(packet->duration * 1000));
+                          llrint(packet->pts * 1000),
+                          llrint(packet->duration * 1000));
     }
 }

